server/routes/add-any-alert/schemas.ts

```diff
@@ -0,0 +1,96 @@
+import { Request } from 'express'
+import z, { RefinementCtx } from 'zod'
+import { isBefore, parseISO } from 'date-fns'
+import {
+  createSchema,
+  validateAndTransformReferenceData,
+  validateTransformDateInRange,
+  validateTransformOptionalDate,
+} from '../../middleware/validationMiddleware'
+import AlertsApiClient from '../../data/alertsApiClient'
+import PrisonerSearchApiClient from '../../data/prisonerSearchApiClient'
+
+export const schemaFactory =
+  (alertsApiClient: AlertsApiClient, prisonerSearchApiClient: PrisonerSearchApiClient) => async (req: Request) => {
+    const types = await alertsApiClient.retrieveAlertTypes(req.middleware.clientToken, true)
+
+    const alertTypeMap = new Map(
+      types.map(type => {
+        const { alertCodes, ...rest } = type
+        return [type.code, rest]
+      }),
+    )
+
+    const alertCodeMap = new Map(types.flatMap(type => type.alertCodes).map(code => [code.code, code]))
+
+    return createSchema({
+      prisonNumber: z
+        .string()
+        .regex(/^[A-Z]{1}[0-9]{4}[A-Z]{2}$/, 'Enter a prison number in the format A1234CD')
```

Comment:
This should be case insensitive. I believe prisoner search is also case insensitive, but if not, we should capitalise it before sending

Reply:
prisoner search isn't case insensitive. we'll accept lowercase characters and capitalise them in our api call.